File: 32_01_PrintTreeFromTopToBottom/PrintTreeFromTopToBottom.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def PrintFromTopToBottom(self, root):
        L = []
        if not root:
            return L
        queue = [root]
        while queue:
            root = queue.pop(0)
            L.append(root.val)
            if root.left:
                queue.append(root.left)
            if root.right:
                queue.append(root.right)
        return L


t1 = TreeNode(8)
t2 = TreeNode(6)
t3 = TreeNode(10)
t1.left = t2
t1.right = t3
s = Solution()
print(s.PrintFromTopToBottom(t1))

File: 32_01_PrintTreeFromTopToBottom/PrintTreeFromTopToBottom.cpp
#include <iostream>
#include <vector>
#include <string>
#include <deque>
#include <set>
#include <map>
#include <queue>
#include <stack>
#include <algorithm>
#include <cstring>
using namespace std;

struct TreeNode
{
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

class Solution
{
  public:
    vector<int> PrintFromTopToBottom(TreeNode *root)
    {
        queue<TreeNode *> q;
        vector<int> v;
        if (root == nullptr)
            return v;
        q.push(root);
        while (!q.empty())
        {
            root = q.front();
            v.push_back(root->val);
            if (root->left)
                q.push(root->left);
            if (root->right)
                q.push(root->right);
            q.pop();
        }
        return v;
    }
};